### ISSUE TITLE
After installing the latest flutter version, my pub get started to timeout on BOTH computers. I have work to do. Please help.

### ISSUE THREAD

**Issue by fernandasdias** [flutter/flutter#121608]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Upgrade flutter to the latest version
Flutter doctor 

HTTP host "https://pub.dev" is not reachable. Reason: An error occurred while checking the HTTP host: O tempo limite do semáforo expirou.

### Actual results

Doctor summary (to see all details, run flutter doctor -v):
[!] Flutter (Channel stable, 3.7.5, on macOS 12.6.3 21G419 darwin-x64, locale en-US)
    ! Warning: `dart` on your path resolves to /usr/local/Cellar/dart/2.17.1/libexec/bin/dart, which is not inside your current Flutter SDK checkout at
      /Users/dias/Documents/flutter. Consider adding /Users/dias/Documents/flutter/bin to the front of your path.
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] VS Code (version 1.75.1)
[✓] VS Code (version 1.58.2)
[✓] Connected device (2 available)
[!] HTTP Host Availability
    ✗ HTTP host "https://pub.dev/" is not reachable. Reason: An error occurred while checking the HTTP host: Handshake error in client

### Logs

<details>
<summary>Logs</summary>

```console

<!-- Paste your logs here -->
a
```

</details>

### Flutter Doctor output

<details>
<summary>Doctor output</summary>

```console

<!-- Paste your output here -->

Doctor summary (to see all details, run flutter doctor -v):
[!] Flutter (Channel stable, 3.7.5, on macOS 12.6.3 21G419 darwin-x64, locale en-US)
    ! Warning: `dart` on your path resolves to /usr/local/Cellar/dart/2.17.1/libexec/bin/dart, which is not inside your current Flutter SDK checkout at
      /Users/dias/Documents/flutter. Consider adding /Users/dias/Documents/flutter/bin to the front of your path.
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] VS Code (version 1.75.1)
[✓] VS Code (version 1.58.2)
[✓] Connected device (2 available)
[!] HTTP Host Availability
    ✗ HTTP host "https://pub.dev/" is not reachable. Reason: An error occurred while checking the HTTP host: Handshake error in client

</details>

**Comment by fernandasdias** [comment]
> ### Is there an existing issue for this?
> * [x]  I have searched the [existing issues](https://github.com/flutter/flutter/issues)
> * [x]  I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)
> 
> ### Steps to reproduce
> Upgrade flutter to the latest version Flutter doctor
> 
> HTTP host "https://pub.dev" is not reachable. Reason: An error occurred while checking the HTTP host: O tempo limite do semáforo expirou.
> 
> ### Actual results
> Doctor summary (to see all details, run flutter doctor -v): [!] Flutter (Channel stable, 3.7.5, on macOS 12.6.3 21G419 darwin-x64, locale en-US) ! Warning: `dart` on your path resolves to /usr/local/Cellar/dart/2.17.1/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/dias/Documents/flutter. Consider adding /Users/dias/Documents/flutter/bin to the front of your path. [✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3) [✓] Xcode - develop for iOS and macOS (Xcode 14.2) [✓] Chrome - develop for the web [✓] Android Studio (version 2021.2) [✓] VS Code (version 1.75.1) [✓] VS Code (version 1.58.2) [✓] Connected device (2 available) [!] HTTP Host Availability ✗ HTTP host "https://pub.dev/" is not reachable. Reason: An error occurred while checking the HTTP host: Handshake error in client
> 
> ### Logs
> Logs
> ### Flutter Doctor output
> Doctor output
> ```
> 
> 
> Doctor summary (to see all details, run flutter doctor -v):
> [!] Flutter (Channel stable, 3.7.5, on macOS 12.6.3 21G419 darwin-x64, locale en-US)
>     ! Warning: `dart` on your path resolves to /usr/local/Cellar/dart/2.17.1/libexec/bin/dart, which is not inside your current Flutter SDK checkout at
>       /Users/dias/Documents/flutter. Consider adding /Users/dias/Documents/flutter/bin to the front of your path.
> [✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
> [✓] Xcode - develop for iOS and macOS (Xcode 14.2)
> [✓] Chrome - develop for the web
> [✓] Android Studio (version 2021.2)
> [✓] VS Code (version 1.75.1)
> [✓] VS Code (version 1.58.2)
> [✓] Connected device (2 available)
> [!] HTTP Host Availability
>     ✗ HTTP host "https://pub.dev/" is not reachable. Reason: An error occurred while checking the HTTP host: Handshake error in client
> 
> </details>
> ```

Plus, my path is 100% correct. I don't know why or where this path resolves to /usr/local/Cellar/dart/2.17.1/libexec/bin/dart is coming from. I couldn't find it

**Comment by darshankawar** [comment]
> I don't know why or where this path resolves to /usr/local/Cellar/dart/2.17.1/libexec/bin/dart is coming from. I couldn't find it

You may follow-up in https://github.com/flutter/flutter/issues/121539 for further updates on this warning generated.

Regarding HTTP host availability, do you consistently get the said error ? It might be a glitch in the network connection while upgrading was happening.

**Comment by fernandasdias** [comment]
Hi, 

Indeed it was something on my network provider. I still don't know what was, but they fixed it. By the way another solution was to turn on my VPN. Thanks a lot
